### DISCUSSION TITLE
Is the new Metadata api comprehensive?

### DISCUSSION THREAD

**Discussion by aretrace** [vercel/next.js#46461] [Help] [answered]
Does it support properties like `user-scalable` on the viewport meta tag?

**Comment by ciruz** [comment, marked as answer]
You can set a lot of informations, take a look at the interface.

https://github.com/vercel/next.js/blob/5792533781e6d6fdf10d5b400d8e97d25bd5edc3/packages/next/src/lib/metadata/types/metadata-interface.ts

It's also possible to set properties like viewport. https://github.com/vercel/next.js/blob/5792533781e6d6fdf10d5b400d8e97d25bd5edc3/packages/next/src/lib/metadata/types/metadata-interface.ts#L159

Even all kinds of OpenGraph Informations https://github.com/vercel/next.js/blob/5792533781e6d6fdf10d5b400d8e97d25bd5edc3/packages/next/src/lib/metadata/types/opengraph-types.ts

**Reply by aretrace** [reply]
It is not explicitly typed:
https://github.com/vercel/next.js/blob/5792533781e6d6fdf10d5b400d8e97d25bd5edc3/packages/next/src/lib/metadata/types/extra-types.ts#L53-L62
I understand this is done for accessibility reasons but perhaps this could be mentioned in the docs for codebases that use typescript.

**Comment by icyJoseph** [comment]
Yes.

According to the `Metadata` type definition:

```ts
    /**
     * The viewport setting for the document.
     * @example
     * ```tsx
     * "width=device-width, initial-scale=1"
     * <meta name="viewport" content="width=device-width, initial-scale=1" />
     *
     * { width: "device-width", initialScale: 1 }
     * <meta name="viewport" content="width=device-width, initial-scale=1" />
     * ```
     */
    viewport?: null | string | Viewport;
```
